org.civicrm.reqangular/src/common/angular-date/src/services/DateFactory.js

```diff
@@ -20,13 +23,33 @@ function DateFactory() {
                 ];
             }
 
-            if(typeof strict === 'undefined'){
+            if (typeof strict === 'undefined') {
                 strict = true;
             }
 
             return moment(dateString, format, strict);
+        },
+
+        /**
+         * @description Returns Current Date Format
+         * @returns {string}
+         */
+        getDateFormat: function () {
+            return this.dateFormat;
+        },
+
+        /**
+         * @description Fetches date format setting from
+         * @returns {Promise}
+         */
+        fetchDateFormatFromSettings: function(){
```

Comment:
Why having two methods for the same thing (`getDateFormat` and `fetchDateFormatFromSettings`)? Can't you have the same method that checks if the format has already been stored in a property and, if not, then calls the service?
```js
getDateFormat: function () {
    if (!this.dateFormat) {
        // get the date format
    }
```